Source/WTF/wtf/text/WTFString.h

```diff
@@ -512,6 +514,18 @@ inline void String::clearImplIfNotShared()
         m_impl = nullptr;
 }
 
+template<typename Func>
+inline Expected<std::invoke_result_t<Func, Span<const char>>, UTF8ConversionError> String::tryGetUtf8ForRange(const Func& function, unsigned offset, unsigned length, ConversionMode mode) const
+{
+    ASSERT(offset <= this->length());
+    ASSERT(offset + length <= this->length());
+    if (!m_impl) {
+        const char* const emptyString = "";
+        return function(Span<const char> { emptyString, emptyString });
```

Comment:
Can we write just `Span` here or does it not deduce the type?

Reply:
Changed.